keyGenerartor/src/base/KeyGenerator.java

```diff
@@ -4,12 +4,12 @@
 
 public class KeyGenerator {
 
-	public static String[] Generator(Integer keyLenght, String currentDictionary) {
-		String[] key = new String[keyLenght];
+	public static String Generator(Integer keyLenght, String currentDictionary) {
+		String key = "";
 
 		for (int index = 0; index < keyLenght; index++) {
 			Character letter = currentDictionary.charAt((new Random()).nextInt(currentDictionary.length()));
-			key[index] = letter.toString();
+			key += letter.toString();
```

Comment:
toString is not necessary with the last changes I think:
```suggestion
			key += letter;
```